### PR TITLE
 Automate and run complete set of NEML regression tests

### PR DESCRIPTION
**Pull request by bwspenc**
ref #197

Update NEML submodule
Remove the .xml and .csv files for the NEML tests and add a script that sets them up based on NEML_DIR

### CODE REVIEW

File: test/tests/neml_regression/neml_regression.i [hunk 1]
```diff
@@ -20,16 +18,19 @@
...
+  [cube]
+    type = GeneratedMeshGenerator
+    dim = 3
+  []
 []
 
 [Modules/TensorMechanics/Master]
   [all]
+    strain = SMALL
+    incremental = true
```
Comment (dschwen) [inline test/tests/neml_regression/neml_regression.i:33]: Why incremental?

Comment (bwspenc) [reply test/tests/neml_regression/neml_regression.i:33]: I think those are both the defaults, actually. It runs the same without them being set.

Comment (dschwen) [reply test/tests/neml_regression/neml_regression.i]: `SMALL` is the defsult. If not supplied `incremental` is set to `false` for small strain. You are selecting an incremental small strain formulation, while we usually use total small strain.

Comment (dschwen) [reply test/tests/neml_regression/neml_regression.i]:
```
  // determine if incremental strains are to be used
  if (isParamValid("incremental"))
  {
    const bool incremental = getParam<bool>("incremental");
    if (!incremental && _strain == Strain::Small)
      _strain_and_increment = StrainAndIncrement::SmallTotal;
    else if (!incremental && _strain == Strain::Finite)
      _strain_and_increment = StrainAndIncrement::FiniteTotal;
    else if (incremental && _strain == Strain::Small)
      _strain_and_increment = StrainAndIncrement::SmallIncremental;
    else if (incremental && _strain == Strain::Finite)
      _strain_and_increment = StrainAndIncrement::FiniteIncremental;
    else
      mooseError("Internal error");
  }
  else
  {
    if (_strain == Strain::Small)
      _strain_and_increment = StrainAndIncrement::SmallTotal;  <------------------ DEFAULT FOR SMALL
    else if (_strain == Strain::Finite)
      _strain_and_increment = StrainAndIncrement::FiniteIncremental;
    else
      mooseError("Internal error");
  }
```

Comment (bwspenc) [reply test/tests/neml_regression/neml_regression.i]: Well, now I'm puzzled. I would have thought that these would require incremental.

Comment (bwspenc) [reply test/tests/neml_regression/neml_regression.i]: Well, looks like the NEML stress calculator actually only uses the total strain, so it doesn't make any difference how that is set.

**Review by dschwen — APPROVED** [review]

**Pull request merged**